### PR TITLE
Make cert-manager + Vault workflow more configurable

### LINKED ISSUE #9: Make cert-manager + Vault workflow more configurable
At the moment users can configure the path of  kubeconfig.

It would also be good if other options could be made configurable:
- cert-manager version to install
- Vault version to install
- root domain of the Vault PKI
- flags to pass to cert-manager components
- whether `ClusterIssuer` or an `Issuer` (in a particular namespace) gets created
...

### PR DESCRIPTION
**Pull request by lonelyCZ**
Signed-off-by: lonelyCZ <531187475@qq.com>

Fixes #9

### CODE REVIEW

**Comment by lonelyCZ** [comment]
There are some warnings outputed refer [#22004](https://github.com/hashicorp/terraform/issues/22004), I don't find an easy way to silience it and only set `-compact-warnings` to reduce the outputs. There is [a way in terragrunt] (https://github.com/gruntwork-io/terragrunt/issues/1395#issuecomment-754251066) recommended to deal it, do we need to use it?

```
Warnings:

- Value for undeclared variable
- Value for undeclared variable

To see the full warning notes, run Terraform without -compact-warnings.
```

File: cm-vault/cm-config/variables.tf [hunk 1]
```diff
@@ -2,3 +2,8 @@ variable "kubeconfig_path" {
   type    = string
   default = "~/.kube/config"
 }
+
+variable "vault_pki_root_domain" {
+  type    = string
+  default = "cert-manager.io"
+}
```
Comment (irbekrm) [inline cm-vault/cm-config/variables.tf]: Nit: newline

File: cm-vault/cm-install/variables.tf [hunk 2]
```diff
@@ -2,3 +2,8 @@ variable "kubeconfig_path" {
   type    = string
   default = "~/.kube/config"
 }
+
+variable "cm_version" {
+  type    = string
+  default = "v1.8.0"
```
Comment (irbekrm) [inline cm-vault/cm-install/variables.tf]: Suggestion: bump to the latest patch version, which is currently `v1.8.2` https://github.com/cert-manager/cert-manager/releases/tag/v1.8.2

File: cm-vault/common.tfvars [hunk 3]
```diff
@@ -1,3 +1,13 @@
...
+kubeconfig_path = "~/.kube/config"
+
+# Cert-manger version to install. Default is "v1.8.0"
+cm_version = "v1.8.0"
+
+# Vault version to install. Default is "0.20.1" that is Vault Helm version corresponding to Vault version "1.10.3".
+# You can check the mapping between the Vault Helm version and the Vault version by executing `helm search repo hashicorp/vault -l`
+vault_version = "0.20.1"
+
+# The root domain of the Vault PKI. Default is ""
```
Comment (irbekrm) [inline cm-vault/common.tfvars]: I think the default is "cert-manager.io" not ""?

**Review by irbekrm — CHANGES_REQUESTED** [review]
Thanks @lonelyCZ !

Good work on this. I've added a few small nits, otherwise happy to lgtm 👍🏼

**Comment by lonelyCZ** [comment]
Thanks for your reviews! I have updated it, please review it again.

/assign @irbekrm

**Comment by irbekrm** [comment]
Thank you @lonelyCZ !

/lgtm
/approve

**Pull request merged**